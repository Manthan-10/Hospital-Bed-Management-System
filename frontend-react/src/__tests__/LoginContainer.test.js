import React from 'react';
import Enzyme,{ shallow, mount } from 'enzyme';
import LoginContainer from '../LoginContainer';
import Adapter from 'enzyme-adapter-react-16';
import * as reactRouterDom from "react-router-dom";
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() })

/**
 * Testing for UI of Login page
 * @author MN097789
 */
 

 const mockNavigate = jest.fn();

 jest.mock("react-router-dom", () => ({
 
   useSelector: jest.fn(),
 
   useNavigate: () => mockNavigate,
 
 }));
 const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");

 
 describe('DefaultContainer UI Test Suite', () => {

  

  const initialState = { output: 10 };
    const mockStore = configureStore();
    let store;

  test('Checks if it renders correctly', () => {
    store = mockStore(initialState);
    const tree = renderer.create(<Provider store={store}><LoginContainer /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });


 
});