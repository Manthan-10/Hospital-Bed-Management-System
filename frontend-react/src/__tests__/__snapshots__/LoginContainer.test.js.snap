// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultContainer UI Test Suite Checks if it renders correctly 1`] = `null`;
